emlog.c

```diff
@@ -256,10 +258,16 @@ static char * read_from_emlog(struct emlog_info * einfo, size_t * length,
     int bytes_copied = 0, n, start_point;
     size_t remaining;
 
+    read_lock(&einfo->rwlock);
+
+    if (emlog_debug)
+        pr_debug("\nLength: %zu\nOffset: %lld\neinfo->offset: %lld\nEMLOG_FIRST_EMPTY_BYTE: %lld\neinfo->read_point: %zu\neinfo->write_point: %zu\n", *length, *offset, einfo->offset, EMLOG_FIRST_EMPTY_BYTE(einfo), einfo->read_point, einfo->write_point);
+
     /* is the user trying to read data that has already scrolled off? */
     if (*offset < einfo->offset)
         *offset = einfo->offset;
-
+    if (emlog_debug)
+        pr_debug("New Offset: %lld\n", *offset);
     /* is the user trying to read past EOF? */
     if (*offset >= EMLOG_FIRST_EMPTY_BYTE(einfo))
         return NULL;
```

Comment:
Missing `read_unlock()`?

Reply:
Line 300. Thanks.

Reply:
No, I mean in case of `return NULL` (new line 273), there is no `read_unlock()`, right?